Evgeniya_Bogdanova/js/scripts.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Created by John Doe on 17.10.2017.
+ */
+document.onreadystatechange = function () {
+  if (document.readyState === 'interactive') {
+    initCalculator();
+  }
+};
+
+function initCalculator() {
+  var
+    calculator = document.getElementById('calculator'),
+    first = document.getElementById('first-number'),
+    second = document.getElementById('second-number'),
+    action = document.getElementById('action');
+
+  calculator.onsubmit = function(event) {
+    var
+      firstNumber = parseFloat(first.value),
+      secondNumber = parseFloat(second.value),
+      actionType = action.value.trim(),
+      result = calculate(firstNumber, secondNumber, actionType);
+
+    event.preventDefault(); 
+    
+    if (isNaN(result) || result === undefined) return;
```

Comment:
Very clever!
But because of  `default` section in `switch..case` - **result** is never be `undefined`.